enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,276 @@
+       that can be used to control and provision it independently of
+       its host.
+
+    3. The NIC implements SR-IOV in such a way that we can connect
+       pods on the host directly to an OVN bridge on the NIC.
+
+- Design an appropriate architecture for an environment containing a
+  mix of x86_64 hosts with "dumb" NICs, x86_64 hosts with Smart NICs,
+  and the ARM-based Smart NICs themselves. (As discussed below, such
+  an environment may involve one or more OCP clusters.)
+
+- Define how OCP installs and upgrades will work in such an
+  environment.
+
+This enhancement does not cover what we actually _do_ with the NICs
+after they are provisioned. The primary use case of OVN offload is
+covered by "[Smart NIC OVN Offload]". It is possible that we may also
+want to allow customers to move additional network, security, and
+monitoring functionality from the CPU to the DPU in the future.
+
+[Smart NIC OVN Offload]: https://github.com/openshift/enhancements/pull/732
+
+### Non-Goals
+
+- Support for ASIC- or FPGA-based Smart NICs.
+
+- Support for ARM-based Smart NICs where we would not be able to run
+  RHCOS on the NIC (eg, because the NIC requires proprietary drivers).
+
+## Design Details
+
+### Clusters Will Contain a Mix of Smart and "Dumb" NICs
+
+OVN Kubernetes is not currently capable of mixing offloaded and
+non-offloaded networking on the same node, so every pod on a node
+containing a Smart NIC must have offloaded networking, which, in the
+current architecture, means it must explicitly _request_ offloaded
+networking. In particular, this means that most pods created by
+OpenShift operators cannot be deployed on a node with a Smart NIC, and
+so all masters and at least a few workers will need to have "dumb"
+NICs.
+
+(While the exact architectural constraints may change in the future,
+it is likely that some customers will always want the possibility of
+only having Smart NICs on a subset of nodes.)
+
+### Overall Cluster Architecture
+
+There are several possible overall architectures:
+
+  - The Smart NICs could be nodes within the same OCP cluster as their
+    hosts. We would need to use some combination of labels and taints
+    to distinguish "real" nodes from Smart NIC nodes. (Among other
+    things, the Smart NICs would be their own `MachineConfigPool`,
+    separate from the masters and workers.) This may require changes
+    to tolerations and node selectors on existing DaemonSets, both
+    within OCP and in third-party operators or customer workloads (and
+    it may turn out that this approach is unworkable for this reason).
+
+  - The Smart NICs could be part of a second cluster running in
+    parallel with the primary cluster, with the control plane of the
+    NIC cluster running as pods within the primary cluster.
+
+  - The Smart NICs could be part of a second cluster which is
+    completely separate from the primary cluster, with its own master
```

Comment:
It is expected that there will be some kinds of control plane communication between two separate clusters to exchange necessary information for deployment and running workload. It would not be totally separated and we perhaps need to provide the management of such info exchanges.